src/Microsoft.NetCore.Analyzers/Core/Runtime/DoNotUseEnumerableMethodsOnIndexableCollectionsInsteadUseTheCollectionDirectly.Fixer.cs

```diff
@@ -77,14 +79,48 @@ private async Task<Document> UseCollectionDirectly(Document document, TextSpan s
             }
 
             var generator = SyntaxGenerator.GetGenerator(document);
-            var indexNode = generator.LiteralExpression(0);
-            var elementAccessNode = generator.ElementAccessExpression(collectionSyntax.WithoutTrailingTrivia(), indexNode)
-                .WithTrailingTrivia(invocationNode.GetTrailingTrivia());
 
-            var newRoot = root.ReplaceNode(invocationNode, elementAccessNode);
+            var elementAccessNode = GetReplacementNode(generator, collectionSyntax, methodName);
+            if (elementAccessNode == null)
+            {
+                return document;
+            }
+
+            var newRoot = root.ReplaceNode(invocationNode, elementAccessNode.WithTrailingTrivia(invocationNode.GetTrailingTrivia()));
             return document.WithSyntaxRoot(newRoot);
         }
 
+        private static SyntaxNode GetReplacementNode(SyntaxGenerator generator, SyntaxNode collectionSyntax, string methodName)
+        {
+            var collectionSyntaxNoTrailingTrivia = collectionSyntax.WithoutTrailingTrivia();
+
+            if (methodName == "First")
+            {
+                var zeroLiteral = generator.LiteralExpression(0);
+                return generator.ElementAccessExpression(collectionSyntaxNoTrailingTrivia, zeroLiteral);
+            }
+
+            if (methodName == "Last")
+            {
+                // TODO: Handle C# 8 index expression (and vb.net equivalent if any)
+                // TODO: Handle cases were `collectionSyntax` is an invocation. We would need to create some intermediate variable.
+                var countMemberAccess = generator.MemberAccessExpression(collectionSyntaxNoTrailingTrivia, "Count");
+                var oneLiteral = generator.LiteralExpression(1);
+
+                // The SubstractExpression method will wrap left and right in parenthesis but those will be automatically removed later on
+                var substraction = generator.SubtractExpression(countMemberAccess, oneLiteral);
+                return generator.ElementAccessExpression(collectionSyntaxNoTrailingTrivia, substraction);
+            }
+
+            if (methodName == "Count")
+            {
+                return generator.MemberAccessExpression(collectionSyntaxNoTrailingTrivia, "Count");
```

Comment:
You should add a defensive check that the type of the receiver has a property named `Count`. If not, you should bail out.